sdk/core/Azure.Core/src/SerializableOptions.cs

```diff
@@ -0,0 +1,26 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+namespace Azure
+{
+    /// <summary>
+    /// TODO
+    /// </summary>
+    public class SerializableOptions
+    {
+        /// <summary>
+        /// TODO
+        /// </summary>
+        public bool IncludeReadOnlyProperties { get; set; } = true;
+
+        /// <summary>
+        /// TODO
+        /// </summary>
+        public bool HandleAdditionalProperties { get; set; } = true;
+
+        /// <summary>
+        /// TODO
+        /// </summary>
+        public bool PrettyPrint { get; set; } = false;
```

Comment:
This is rather specific. Instead, should we may let users pass a `System.Text.Json.JsonSerializerOptions`? For example, what if they want 2 spaces to indent instead of 4? Or tabs instead of spaces? If we're going to make pretty printing possible for serialization, we should consider being flexible now instead of doing it later and having this then-confusing property.

Alternatively, what about serializing to a `Utf8Writer` instead of a `Stream`, in which case they can provide a formatting options completely outside our API with whatever options they want?

Reply:
One of the goals here is not pin us to a specific underlying serialization technology and give us the freedom to swap this out as needed under the covers so we wouldn't want to take these types on the public signature.

Reply:
Then at the very least we should probably replicate more of STJ's options. What does "pretty print" even mean? See my examples in my comment for ways this could vary.

Reply:
We definitely don't want to duplicate SJT's options for the reasons listed here https://github.com/Azure/azure-sdk-for-net/pull/35742#discussion_r1176960613